src/MongoDB/Manager.c

```diff
@@ -349,6 +349,26 @@ static PHP_METHOD(Manager, __construct)
 	}
 } /* }}} */
 
+/* {{{ proto MongoDB\Driver\ClientEncryption MongoDB\Driver\Manager::clientEncryption(array $options)
+   Return a ClientEncryption instance */
+static PHP_METHOD(Manager, clientEncryption)
+{
+	php_phongo_manager_t*          intern;
+	php_phongo_clientencryption_t* clientencryption;
+	zval*                          options;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "a", &options) == FAILURE) {
```

Comment:
Creating a ClientEncryption object without parameters causes an exception:
> Key vault client and namespace option required

The `null` and `IS_ARRAY` checks for `options` in `phongo_clientencryption_opts_from_zval` to prevent segmentation faults if the method is called with a non-array `options`. I can change this to an exception instead of silently returning empty options, but allowing `null` to be passed to the `clientEncryption` method is not a good idea IMO.

Reply:
Ah, I didn't realize that two options were necessary. Technically, `keyVaultClient` isn't necessary since it defaults to the Manager's client, but `keyVaultNamespace` would definitely be required so I suppose the minimum value you could pass as the options here is `['keyVaultNamespace' => '...']`.

From an API standpoint, I wonder if `keyVaultNamespace` should be a required parameter of this method, which you then use to call `mongoc_client_encryption_opts_set_keyvault_namespace`. Is `keyVaultNamespace` also a required key in the `autoEncryption` array within `$driverOptions`? Typically, anything required would be its own parameter and `$options` would contain strictly optional values. But if you'd prefer to just take everything in one array here for consistency with `autoEncryption` (obviously we have no other approach to take there, since it's all within `$driverOptions`), then that works for me.

If we keep `keyVaultNamespace` in the options array, `zend_parse_parameters` can continue to use `a` since an array will certainly be required to specify that required option.

The only reason I suggested `|a!` is because it'd be consistent for all other options arrays in PHPC. `ZEND_ARG_ARRAY_INFO(0, options, 1)` appears numerous places in the driver. Unfortunately, I don't recall exactly why we accepted null there instead of just using `|a` and requiring an empty array at a minimum. [Registering and Using PHP Functions](http://www.phpinternalsbook.com/php7/extensions_design/php_functions.html) in the PHP 7 internals book states that `zend_parse_parameters` doesn't even touch the output variable if it wasn't provided by the caller, which is probably why we initialize all of these zval pointers to null ourselves -- that seems completely unrelated to our use of the `!` operator to accept `null` PHP values.

The alternative

Reply:
https://jira.mongodb.org/browse/PHPC-1531

Reply:
I originally had `keyVaultNamespace` as separate argument, but put it into the options array for consistency with other encryption options, including those for `autoEncryption`.